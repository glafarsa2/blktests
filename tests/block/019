#!/bin/bash
#
# Do disable PCI device while doing I/O to it
#
# Copyright (C) 2018 Keith Busch <keith.busch@intel.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

. tests/block/rc

DESCRIPTION="break PCI link device while doing I/O"
QUICK=1

requires() {
	_have_fio && _have_program setpci
}

device_requires() {
	_test_dev_is_pci && _test_dev_in_hotplug_slot
}

test_device() {
	echo "Running ${TEST_NAME}"

	local parent

	parent="$(_get_pci_parent_from_blkdev)"

	# start fio job
	_run_fio_rand_io --filename="$TEST_DEV" --time_based --runtime=10 \
		--continue_on_error=io 2> /dev/null &
	sleep 5

	# masks  the slot's link disable bit to 'on'
	setpci -s "${parent}" CAP_EXP+10.w=10:10
	sleep 5

	# masks the slot's link disable bit back to 'off'
	setpci -s "${parent}" CAP_EXP+10.w=00:10
	sleep 5

	wait

	echo "Test complete"
}
