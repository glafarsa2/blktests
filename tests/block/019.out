Running block/019
Test complete
